src/cov.jl

```diff
@@ -48,76 +49,60 @@ function scattermat end
-        mean == 0 ? scattermat_zm(x, vardim) :
-        mean == nothing ? scattermat_zm(x .- Base.mean(x, vardim), vardim) :
-        scattermat_zm(x .- mean, vardim)
-    end
-
-    function scattermat(x::DenseMatrix, wv::WeightVec; mean=nothing, vardim::Int=1)
-        mean == 0 ? scattermat_zm(x, wv, vardim) :
-        mean == nothing ? scattermat_zm(x .- Base.mean(x, wv, vardim), wv, vardim) :
-        scattermat_zm(x .- mean, wv, vardim)
-    end
+scattermatm(x::DenseMatrix, mean, vardim::Int=1) =
+    scattermat_zm(x .- mean, vardim)
 
-    ## weighted cov
-    Base.cov(x::DenseMatrix, wv::WeightVec; mean=nothing, vardim::Int=1) =
-        scale!(scattermat(x, wv; mean=mean, vardim=vardim), inv(sum(wv)))
+scattermatm(x::DenseMatrix, mean, wv::AbstractWeights, vardim::Int=1) =
+    scattermat_zm(x .- mean, wv, vardim)
 
-    function mean_and_cov(x::DenseMatrix; vardim::Int=1)
-        m = mean(x, vardim)
-        return m, Base.covm(x, m; vardim=vardim)
-    end
-    function mean_and_cov(x::DenseMatrix, wv::WeightVec; vardim::Int=1)
-        m = mean(x, wv, vardim)
-        return m, Base.cov(x, wv; mean=m, vardim=vardim)
-    end
-else
-    scattermatm(x::DenseMatrix, mean, vardim::Int=1) =
-        scattermat_zm(x .- mean, vardim)
+scattermat(x::DenseMatrix, vardim::Int=1) =
+    scattermatm(x, Base.mean(x, vardim), vardim)
 
-    scattermatm(x::DenseMatrix, mean, wv::WeightVec, vardim::Int=1) =
-        scattermat_zm(x .- mean, wv, vardim)
+scattermat(x::DenseMatrix, wv::AbstractWeights, vardim::Int=1) =
+    scattermatm(x, Base.mean(x, wv, vardim), wv, vardim)
 
-    scattermat(x::DenseMatrix, vardim::Int=1) =
-        scattermatm(x, Base.mean(x, vardim), vardim)
+## weighted cov
+Base.covm(x::DenseMatrix, mean, wv::AbstractWeights, vardim::Int=1;
+          corrected::DepBool=nothing) =
+    scale!(scattermatm(x, mean, wv, vardim), varcorrection(wv, depcheck(:covm, corrected)))
 
-    scattermat(x::DenseMatrix, wv::WeightVec, vardim::Int=1) =
-        scattermatm(x, Base.mean(x, wv, vardim), wv, vardim)
 
-    ## weighted cov
-    Base.covm(x::DenseMatrix, mean, wv::WeightVec, vardim::Int=1) =
-        scale!(scattermatm(x, mean, wv, vardim), inv(sum(wv)))
+Base.cov(x::DenseMatrix, wv::AbstractWeights, vardim::Int=1; corrected::DepBool=nothing) =
+    Base.covm(x, Base.mean(x, wv, vardim), wv, vardim; corrected=depcheck(:cov, corrected))
 
-    Base.cov(x::DenseMatrix, wv::WeightVec, vardim::Int=1) =
-        Base.covm(x, Base.mean(x, wv, vardim), wv, vardim)
 
-    function mean_and_cov(x::DenseMatrix, vardim::Int=1)
-        m = mean(x, vardim)
-        return m, Base.covm(x, m, vardim)
-    end
-    function mean_and_cov(x::DenseMatrix, wv::WeightVec, vardim::Int=1)
-        m = mean(x, wv, vardim)
-        return m, Base.cov(x, wv, vardim)
-    end
+function mean_and_cov(x::DenseMatrix, vardim::Int=1; corrected::Bool=true)
+    m = mean(x, vardim)
+    return m, Base.covm(x, m, vardim, corrected)
```

Comment:
Shouldn't this be `corrected=corrected`? If so, that means that function wasn't tested at all?